examples/cifar10_cnn_capsule.py

```diff
@@ -0,0 +1,211 @@
+        input_num_capsule = K.shape(u_vecs)[1]
+        u_hat_vecs = K.reshape(u_hat_vecs, (batch_size,
+                                            input_num_capsule,
+                                            self.num_capsule,
+                                            self.dim_capsule))
+        u_hat_vecs = K.permute_dimensions(u_hat_vecs, (0, 2, 1, 3))
+        # final u_hat_vecs.shape = [None, num_capsule,
+        #                           input_num_capsule, dim_capsule]
+
+        b = K.zeros_like(u_hat_vecs[:, :, :, 0])  # shape = [None, num_capsule,
+        #                                                    input_num_capsule]
+        for i in range(self.routings):
+            c = softmax(b, 1)
+            outputs = self.activation(K.batch_dot(c, u_hat_vecs, [2, 2]))
+            if i < self.routings - 1:
+                b = K.batch_dot(outputs, u_hat_vecs, [2, 3])
+
+        return outputs
+
+    def compute_output_shape(self, input_shape):
+        return (None, self.num_capsule, self.dim_capsule)
+
+
+# some parameters
+batch_size = 128
+num_classes = 10
+epochs = 100
+(x_train, y_train), (x_test, y_test) = cifar10.load_data()
+
+x_train = x_train.astype('float32')
+x_test = x_test.astype('float32')
+x_train /= 255
+x_test /= 255
+y_train = utils.to_categorical(y_train, num_classes)
+y_test = utils.to_categorical(y_test, num_classes)
+
+# A common Conv2D model
+input_image = Input(shape=(None, None, 3))
+cnn = Conv2D(64, (3, 3), activation='relu')(input_image)
+cnn = Conv2D(64, (3, 3), activation='relu')(cnn)
+cnn = AveragePooling2D((2, 2))(cnn)
+cnn = Conv2D(128, (3, 3), activation='relu')(cnn)
+cnn = Conv2D(128, (3, 3), activation='relu')(cnn)
+
+'''
+now we reshape it as (batch_size, input_num_capsule, input_dim_capsule)
+then connect a Capsule layer.
+the output of final model is the lengths of 10 Capsule, who dim=16
+the length of Capsule is the proba,
+so the probelm becomes a 10 two-classification problems
+'''
+
+cnn = Reshape((-1, 128))(cnn)
+capsule = Capsule(10, 16, 3, True)(cnn)
+output = Lambda(lambda x: K.sqrt(K.sum(K.square(x), 2)))(capsule)
+model = Model(inputs=input_image, outputs=output)
+
+# we use a margin loss
+model.compile(loss=lambda y_true, y_pred: y_true * K.relu(0.9 - y_pred)**2 +
+              0.25 * (1 - y_true) * K.relu(y_pred - 0.1)**2,
+              optimizer='adam',
+              metrics=['accuracy'])
+model.summary()
+
+# we can compare the perfermace with or without data augmentation
+data_augmentation = True
+
+if not data_augmentation:
+    print('Not using data augmentation.')
+    model.fit(x_train, y_train,
+              batch_size=batch_size,
+              epochs=epochs,
+              validation_data=(x_test, y_test),
+              shuffle=True)
+else:
+    print('Using real-time data augmentation.')
+    # This will do preprocessing and realtime data augmentation:
+    datagen = ImageDataGenerator(
+        featurewise_center=False,  # set input mean to 0 over the dataset
```

Comment:
Everything in the `ImageDataGenerator` is already documented, I'm not 100% sure but I would remove the inline comments.

Reply:
This is copied verbatim from another example. If comments are removed, it'll be inconsistent with other examples, where this is pervasive throughout.